src/engagement_db_to_coda/engagement_db_to_coda.py

```diff
@@ -155,10 +154,40 @@ def _sync_message_to_coda(transaction, engagement_db, coda, coda_config, message
     :type engagement_db: engagement_database.EngagementDatabase
     :param coda: Coda instance to sync the message to.
     :type coda: coda_v2_python_client.firebase_client_wrapper.CodaV2Client
-    :param message_id: Id of the message to sync.
-    :type message_id: str
+    :param coda_config: Coda sync configuration.
+    :type coda_config: src.engagement_db_to_coda.configuration.CodaSyncConfiguration
+    :param last_seen_message: Last seen message, downloaded from the database in a previous call, or None.
+                              If provided, downloads the least recently updated (next) message after this one, otherwise
+                              downloads the least recently updated message in the database.
+    :type last_seen_message: engagement_database.data_models.Message | None
+    :return: The engagement database message that was synced.
+             If there was no new message to sync, returns None.
+    :rtype: engagement_database.data_models.Message | None
     """
-    engagement_db_message = engagement_db.get_message(message_id, transaction=transaction)
+    if last_seen_message is None:
+        messages_filter = lambda q: q \
+            .where("status", "in", [MessageStatuses.LIVE, MessageStatuses.STALE]) \
+            .order_by("last_updated") \
+            .order_by("message_id") \
+            .limit(1)
+    else:
+        # Get the next message modified at or later than the `last_seen_message`, excluding the `last_seen_message`.
+        messages_filter = lambda q: q \
+            .where("status", "in", [MessageStatuses.LIVE, MessageStatuses.STALE]) \
+            .order_by("last_updated") \
+            .order_by("message_id") \
+            .where("last_updated", ">=", last_seen_message.last_updated) \
+            .start_after({"last_updated": last_seen_message.last_updated, "message_id": last_seen_message.message_id}) \
+            .limit(1)
+
+    next_message_results = engagement_db.get_messages(filter=messages_filter, transaction=transaction)
```

Comment:
I think we should change param filter to firestore_filter or sth similar, to differentiate from the inbuilt module.

Reply:
have added as a low priority todo